src/ndiaudiosrc.rs

```diff
@@ -258,24 +258,18 @@ impl ObjectSubclass for NdiAudioSrc {
 
                 let audio_frame: NDIlib_audio_frame_v2_t = Default::default();
 
-                let mut frame_type: NDIlib_frame_type_e = NDIlib_frame_type_e::NDIlib_frame_type_none;
                 unsafe {
-                    while frame_type != NDIlib_frame_type_e::NDIlib_frame_type_audio {
-                        frame_type = NDIlib_recv_capture_v2(
-                            pNDI_recv,
-                            ptr::null(),
-                            &audio_frame,
-                            ptr::null(),
-                            1000,
-                        );
-                        gst_debug!(self.cat, obj: element, "NDI audio frame received: {:?}", audio_frame);
+                    while NDIlib_recv_capture_v2(pNDI_recv, ptr::null(), &audio_frame, ptr::null(), 1000) != NDIlib_frame_type_e::NDIlib_frame_type_audio {
+                        NDIlib_recv_free_audio_v2(pNDI_recv, &audio_frame);
```

Comment:
Looks better than before.
Although, why is there a need to call `NDIlib_recv_free_audio_v2()` if the `recv_capture` function didn't return a `type_audio`?

Reply:
Indeed, I may have overdid it, looks even better now.